src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt

```diff
@@ -152,4 +156,33 @@ object StringUtils {
         builder.append(end)
         return builder.toString()
     }
+
+    // recursively goes through the component and siblings until an action = true
+    fun modifyFirstChatComponent(chatComponent: IChatComponent, action: Predicate<IChatComponent>): Boolean {
+        if (action.test(chatComponent)) {
+            return true
+        }
+        for (sibling in chatComponent.siblings) {
+            if (modifyFirstChatComponent(sibling, action)) {
+                return true
+            }
+        }
+        return false
+    }
+
+    // replaces without breaking any click or hover events (unless that whole text is removed)
+    fun replaceFirstChatText(chatComponent: IChatComponent, toReplace: String, replacement: String): IChatComponent {
+        modifyFirstChatComponent(chatComponent) { component ->
+            if (component is ChatComponentText && component.text.contains(toReplace)) {
+
+                component.text = component.text.replace(toReplace, replacement)
+                true
+            } else {
+                false
```

Comment:
You can remove the if and instead directly use a `return@modifyFirstChatComponent /* your if condition here 
*/`